dataprofiler/profilers/profile_builder.py

```diff
@@ -1927,18 +1954,39 @@ def _update_row_statistics(
                 "Cannot calculate row statistics on data that is" "not a DataFrame"
             )
 
-        self.total_samples += len(data)
-        try:
-            self.hashed_row_dict.update(
-                dict.fromkeys(pd.util.hash_pandas_object(data, index=False), True)
-            )
-        except TypeError:
-            self.hashed_row_dict.update(
-                dict.fromkeys(
-                    pd.util.hash_pandas_object(data.astype(str), index=False), True
-                )
-            )
+        if not self.options.row_statistics.is_enabled:
+            return
 
```

Comment:
consistent with the other updates, should we move this to where the call happens?